exp/api/v1alpha4/conversion_test.go

```diff
@@ -18,14 +18,42 @@ package v1alpha4
 
 import (
 	"testing"
-
+	
+	fuzz "github.com/google/gofuzz"
 	. "github.com/onsi/gomega"
-
-	runtime "k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/api/apitesting/fuzzer"
+	"k8s.io/apimachinery/pkg/runtime"
+	runtimeserializer "k8s.io/apimachinery/pkg/runtime/serializer"
+	
+	"sigs.k8s.io/cluster-api-provider-aws/api/v1alpha4"
 	"sigs.k8s.io/cluster-api-provider-aws/exp/api/v1beta1"
 	utilconversion "sigs.k8s.io/cluster-api/util/conversion"
 )
 
+func fuzzFuncs(_ runtimeserializer.CodecFactory) []interface{} {
+	return []interface{}{
+		AWSMachinePoolFuzzer,
+	}
+}
+
+func AWSMachinePoolFuzzer(obj *AWSMachinePool, c fuzz.Continue) {
+	c.FuzzNoCustom(obj)
+	
+	// AWSMachinePool.Spec.AWSLaunchTemplate.AWSLaunchTemplate.AdditionalSecurityGroups.ARN and AWSMachinePool.Spec.Subnets.ARN have been removed in v1beta1, so setting it to nil in order to avoid v1alpha3 --> v1beta1 --> v1alpha3 round trip errors.
```

Comment:
```suggestion
	// AWSMachinePool.Spec.AWSLaunchTemplate.AWSLaunchTemplate.AdditionalSecurityGroups.ARN and AWSMachinePool.Spec.Subnets.ARN has been deprecated in v1beta1, so setting it to nil in order to avoid v1alpha3 --> v1beta1 --> v1alpha3 round trip errors.
```

Reply:
oh i think this change has to be removed, we can add this change when we actually remove the field.

Reply:
There are some other conversions files as well, which will require the change only when we remove ARN field, I will revert those as well.

Reply:
Done